addons/point_of_sale/models/pos_config.py

```diff
@@ -846,4 +846,4 @@ def _get_payment_method(self, payment_type):
         return False
 
     def _get_special_products(self):
-        return self.env.ref('point_of_sale.product_product_tip', raise_if_not_found=False) or self.env['product.product']
+        return self.env.ref('point_of_sale.product_product_tip', raise_if_not_found=False) and self.env.ref('point_of_sale.product_product_consumable', raise_if_not_found=False) or self.env['product.product']
```

Comment:
I don't think this change is necessary.

Reply:
If I don't do that, discount product is show in the product screen